stan/math/prim/fun/grad_2F1.hpp

```diff
@@ -76,7 +76,7 @@ TupleT grad_2F1_impl_ab(const T1& a1, const T2& a2, const T3& b1, const T_z& z,
   ScalarT inner_diff = 1;
   ScalarArrayT g_current = ScalarArrayT::Zero(3);
 
-  while (inner_diff > precision && k < max_steps) {
+  while ((inner_diff > precision || k < 5) && k < max_steps) {
```

Comment:
Just above this, can we add:

```
int min_steps = 5;
```

then change the conditional to use the named variable? It'll make it easier to parse.